src/api/BrowserTypes.ts

```diff
@@ -5861,3 +5861,738 @@ const typeMap: any = {
+    try {
+        Convert.webConnectionProtocol1HelloToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol2LoadURL(value: any): value is WebConnectionProtocol2LoadURL {
+    try {
+        Convert.webConnectionProtocol2LoadURLToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol3Handshake(value: any): value is WebConnectionProtocol3Handshake {
+    try {
+        Convert.webConnectionProtocol3HandshakeToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol4ValidateAppIdentity(value: any): value is WebConnectionProtocol4ValidateAppIdentity {
+    try {
+        Convert.webConnectionProtocol4ValidateAppIdentityToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol5ValidateAppIdentityFailedResponse(value: any): value is WebConnectionProtocol5ValidateAppIdentityFailedResponse {
+    try {
+        Convert.webConnectionProtocol5ValidateAppIdentityFailedResponseToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol5ValidateAppIdentitySuccessResponse(value: any): value is WebConnectionProtocol5ValidateAppIdentitySuccessResponse {
+    try {
+        Convert.webConnectionProtocol5ValidateAppIdentitySuccessResponseToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocol6Goodbye(value: any): value is WebConnectionProtocol6Goodbye {
+    try {
+        Convert.webConnectionProtocol6GoodbyeToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export function isWebConnectionProtocolMessage(value: any): value is WebConnectionProtocolMessage {
+    try {
+        Convert.webConnectionProtocolMessageToJson(value);
+        return true;
+    } catch (e: any) {
+        return false;
+    }
+}
+
+export type RequestMessage = AddContextListenerRequest | AddEventListenerRequest | AddIntentListenerRequest | BroadcastRequest | ContextListenerUnsubscribeRequest | CreatePrivateChannelRequest | EventListenerUnsubscribeRequest | FindInstancesRequest | FindIntentRequest | FindIntentsByContextRequest | GetAppMetadataRequest | GetCurrentChannelRequest | GetCurrentContextRequest | GetInfoRequest | GetOrCreateChannelRequest | GetUserChannelsRequest | HeartbeatAcknowledgementRequest | IntentListenerUnsubscribeRequest | IntentResultRequest | JoinUserChannelRequest | LeaveCurrentChannelRequest | OpenRequest | PrivateChannelAddEventListenerRequest | PrivateChannelDisconnectRequest | PrivateChannelUnsubscribeEventListenerRequest | RaiseIntentForContextRequest | RaiseIntentRequest;
```

Comment:
These names `RequestMessage` `ResponseMessage` and `EventMessage` probably need to change. I am open to suggestions!

Reply:
Seeing this, there is another way. We already generate 3 types that will only validate against these groupings:

- `AppRequestMessage`, 
- `AgentResponseMessage`,
- `AgentEventMessage` 

and each has a union of the `type` field string  values that apply:  

- `RequestMessageType`, 
- `ResponseMessageType`, 
- `EventMessageType`

E.g. https://github.com/finos/FDC3/blob/6cf70dcac3096b6104417d613ed87516cacd3592/src/api/BrowserTypes.ts#L528-L564

These could also be used to implement a type guard.

Could you add a usage example or two so that we can evaluate what the best (and easiest to maintain) solution is?